core/templates/dev/head/components/state-directives/answer-group-editor/answer-group-editor.directive.ts

```diff
@@ -81,23 +81,43 @@ oppia.directive('answerGroupEditor', [
           ctrl.activeRuleIndex = ResponsesService.getActiveRuleIndex();
           ctrl.editAnswerGroupForm = {};
           ctrl.misconceptionName = null;
-          ctrl.misconceptions = StateEditorService.getMisconceptions();
-
-          var _getTaggedMisconceptionName = function(misconceptionId) {
-            for (var i = 0; i < ctrl.misconceptions.length; i++) {
-              if (
-                ctrl.misconceptions[i].getId() === misconceptionId) {
-                ctrl.misconceptionName = ctrl.misconceptions[i].getName();
+          ctrl.misconceptionsPerSkill =
+            StateEditorService.getMisconceptionsPerSkill();
+
+          var _getTaggedMisconceptionName = function(skillMisconceptionId) {
+            if (skillMisconceptionId !== null &&
+              typeof skillMisconceptionId === 'string') {
```

Comment:
Why do we have to introduce this check? (Are we going to receive any value other than null and string?

Reply:
No there will be no other value, but this is to prevent 'split' raise an error.

Reply:
If we do not expect this to happen, then shouldn't we throw an error if it does happen?

Reply:
Done.